### PR TITLE
docs：add o1datagene_cookbook

### PR DESCRIPTION
**Pull request by zjrwtx**
## Description

Describe your changes in detail.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Wendong-Fan** [comment]
please refer to https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md#pull-request-item-stage, update the PR title

**Comment by zjrwtx** [comment]
> please refer to https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md#pull-request-item-stage, update the PR title

thanks！already change the pr title @Wendong-Fan

File: docs/cookbooks/O1datagen_with_CAMEL_And_Upload_Data_To_Huggingface.ipynb [hunk 1]
```diff
@@ -0,0 +1,975 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "64e2b2cd",
+      "metadata": {
+        "id": "64e2b2cd"
+      },
+      "source": [
+        "### please prepare the qa data like below in json file:\n",
```
Comment (koch3092) [inline docs/cookbooks/O1datagen_with_CAMEL_And_Upload_Data_To_Huggingface.ipynb:263]: Shall we need to prepare a simple sample file?

File: docs/cookbooks/O1datagen_with_CAMEL_And_Upload_Data_To_Huggingface.ipynb [hunk 2]
```diff
@@ -0,0 +1,975 @@
...
+        },
+        "id": "e84fb857",
+        "outputId": "66ecaa1f-d791-47c3-a276-5e5e665b7025"
+      },
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "Enter your HuggingFace username: zjrwtxtechstudio\n",
```
Comment (koch3092) [inline docs/cookbooks/O1datagen_with_CAMEL_And_Upload_Data_To_Huggingface.ipynb:884]: Let's consider using a more general one, or putting a public account in the cookbook.

**Comment by Wendong-Fan** [comment]
from Guohao:
1. naming of the data generation method need to be check
2. add more clear description for the methods used in the data gen

**Pull request merged**